### PR TITLE
Update documentation to reflect modBusTTRegisters changes 

### PR DESCRIPTION
**Pull request by crug80**
Update documentation to reflect modBusTTRegisters changes in climate modbus component (possibility to define different setpoint registers belonging to HVAC Mode, as [PR ](https://github.com/home-assistant/core/pull/106977)into the codebase)

## Proposed change

It updates the ModBus Climates entity in order to reflect the new feature added to have the possibility to manage different setpoint temperature registers belonging to the HVAC mode

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/107600
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by janiversen — CHANGES_REQUESTED** [review]
Reference to the PR in core is missing!

**Comment by crug80** [comment]
> Reference to the PR in core is missing!

Hi @janiversen, thanks for your preliminary review. I've already linked the core PR as you can see
immagine
IS that wrong? Should be linked also elsewhere?

File: source/_integrations/modbus.markdown [hunk 1]
```diff
@@ -617,6 +617,39 @@ climates:
...
+              description: "Target Temperature Register value corresponding to HVAC Dry mode."
+              required: false
+              type: integer
+            state_fan_only_target_temp_register:
+              description: "Target Temperature Register value corresponding to HVAC Fan only mode."
+              required: false
+              type: integer
+            state_heat_cool_target_temp_register:
+              description: "Target Temperature Register value corresponding to HVAC Heat/Cool mode."
+              required: false
```
Comment (janiversen) [inline source/_integrations/modbus.markdown]: This does not seem correct, the map is called "*values", the description of each entry talk about "Register value", but the entries are called "*register", so is it register value or register.

**Review by janiversen — CHANGES_REQUESTED** [review]
confusion on register vs register value....apart from that, it seems to be a lot of added complexity.

File: source/_integrations/modbus.markdown (hunk 1 above)
Comment (crug80) [reply source/_integrations/modbus.markdown]: I understand your point of view. They are registers. If you agree, I change the doc and, of course I change the belonging code renaming the map as hvac_mode_target_temp_registers.

Comment (crug80) [reply source/_integrations/modbus.markdown]: Updated with reference to the new PR

File: source/_integrations/modbus.markdown [hunk 2]
```diff
@@ -768,9 +735,9 @@ climates:
         word_byte:
           description: "Swap word ABCD -> DCBA, **not valid with data types: `int16`, `uint16`**"
     target_temp_register:
-      description: "Register address for target temperature (Setpoint) and default one also in case of hvac_mode_target_temp_reg_values usage."
+      description: "Register address for target temperature (Setpoint). Using a list, multiple registers can be defined for each of the available HVAC Mode. The list has to have a fixed size of 7 registers like the 7 available HVAC Modes, having the following relationship: Register **1: HVAC AUTO mode**; Register **2: HVAC Cool mode**; Register **3: HVAC Dry mode**; Register **4: HVAC Fan only mode**; Register **5: HVAC Heat mode**; Register **6: HVAC Heat Cool mode**; Register **7: HVAC OFF mode**. It is possible to set duplicated values for the modes where the devices has not a related register."
```
Comment (janiversen) [inline source/_integrations/modbus.markdown]: This does not sound correct....can you really define multiple registers for each HVAC mode ?

Which device needs multiple registers to set a single HVAC mode ?

**Review by janiversen — CHANGES_REQUESTED** [review]

File: source/_integrations/modbus.markdown (hunk 2 above)
Comment (crug80) [reply source/_integrations/modbus.markdown]: Ehm.. not really, I forgot to add "one" word. I'll do it. Thanks. 
It should be: `Using a list, multiple registers can be defined **one** for each of the available HVAC Mode`

Comment (janiversen) [reply source/_integrations/modbus.markdown]: Well the new text is not a lot more understandable.....you can define one register pr HVAC mode, no need to talk about multiple registers.

File: source/_integrations/modbus.markdown [hunk 3]
```diff
@@ -735,7 +735,7 @@ climates:
...
+      description: "Register address for target temperature (Setpoint). Using a list, it is possible to define one register for each of the available HVAC Modes. The list has to have a fixed size of 7 registers like the 7 available HVAC Modes, having the following relationship: Register **1: HVAC AUTO mode**; Register **2: HVAC Cool mode**; Register **3: HVAC Dry mode**; Register **4: HVAC Fan only mode**; Register **5: HVAC Heat mode**; Register **6: HVAC Heat Cool mode**; Register **7: HVAC OFF mode**. It is possible to set duplicated values for the modes where the devices has not a related register."
```
Comment (janiversen) [inline source/_integrations/modbus.markdown]:
```suggestion
      description: "Register address for target temperature (Setpoint). Using a list, it is possible to define one register for each of the available HVAC Modes. The list has to have a fixed size of 7 registers corresponding to the 7 available HVAC Modes, as follows: Register **1: HVAC AUTO mode**; Register **2: HVAC Cool mode**; Register **3: HVAC Dry mode**; Register **4: HVAC Fan only mode**; Register **5: HVAC Heat mode**; Register **6: HVAC Heat Cool mode**; Register **7: HVAC OFF mode**. It is possible to set duplicated values for the modes where the devices has not a related register." 
```

**Review by janiversen — CHANGES_REQUESTED** [review]

**Comment by crug80** [comment]
It seems that the PR has been merged and released, but the documentation is not updated.

**Review by janiversen — APPROVED** [review]
LGTM, thanks.

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @crug80 and @janiversen :+1:

**Pull request merged**